edit_course/operations/configure.py

```diff
@@ -356,299 +357,357 @@ def get_build_log(instance):
+                    exercise = lobject.as_leaf_class()
+                    if (
+                        not isinstance(exercise, BaseExercise)
+                        or exercise.submissions.count() == 0
+                    ):
+                        exercise.delete()
+                    else:
+                        lobject.status = LearningObject.STATUS.HIDDEN
+                        lobject.order = 9999
+                        # .parent may have been deleted: only save status and order
+                        lobject.save(update_fields=["status", "order"])
+            if module.id not in seen_modules:
+                if module.learning_objects.count() == 0:
+                    module.delete()
                 else:
-                    lobject.status = LearningObject.STATUS.HIDDEN
-                    lobject.order = 9999
-                    lobject.save()
-        if not module.id in seen_modules:
-            if module.learning_objects.count() == 0:
-                module.delete()
+                    module.status = CourseModule.STATUS.HIDDEN
+                    module.save()
+
+        # Clean up obsolete categories.
+        for category in instance.categories.filter(status=LearningObjectCategory.STATUS.HIDDEN):
+            if category.learning_objects.count() == 0:
+                category.delete()
+
+        if "publish_url" in config:
+            success = False
+            publish_errors = []
+            try:
+                response = aplus_auth.get(config["publish_url"])
+            except ConnectionError as e:
+                publish_errors = [str(e)]
             else:
-                module.status = CourseModule.STATUS.HIDDEN
-                module.save()
-
-    # Clean up obsolete categories.
-    for category in instance.categories.filter(status=LearningObjectCategory.STATUS.HIDDEN):
-        if category.learning_objects.count() == 0:
-            category.delete()
-
-    return errors
+                if response.status_code != 200:
+                    publish_errors = [format_lazy(
+                        _("PUBLISH_RESPONSE_NON_200 -- {status_code}"),
+                        status_code=response.status_code
+                    )]
+
+                if response.text:
+                    try:
+                        publish_errors = json.loads(response.text)
+                    except Exception as e:
+                        publish_errors = [format_lazy(
+                            _("PUBLISH_ERROR_JSON_PARSER_FAILED -- {e}, {text}"),
+                            e=e,
+                            text=response.text
+                        )]
+                    else:
+                        if isinstance(publish_errors, dict):
+                            success = publish_errors.get("success", True)
+                            publish_errors = publish_errors.get("errors", [])
+
+                        if isinstance(publish_errors, list):
```

Comment:
```suggestion
                        elif isinstance(publish_errors, list):
```

Reply:
This is on purpose: `publish_errors = publish_errors.get("errors", [])` above might set it to a list